examples/allocation/zig/testdata/greet.zig

```diff
@@ -0,0 +1,65 @@
+const std = @import("std");
+const allocator = std.heap.page_allocator;
+
+// main is required for Zig to compile to enable WASI.
+pub fn main() anyerror!void {
+    return;
+}
+
+extern "env" fn log(ptr: [*]const u8, size: u32) void;
+
+// _log prints a message to the console using log.
+pub fn _log(message: []const u8) void {
+    log(message.ptr, message.len);
+}
+
+pub export fn malloc(length: usize) ?[*]u8 {
+    const buff = allocator.alloc(u8, length) catch return null;
+    return buff.ptr;
+}
+
+pub export fn free(buf: [*]u8, length: usize) void {
+    allocator.free(buf[0..length]);
+}
+
+pub fn _greeting(name: []const u8) ![]u8 {
+    return try std.fmt.allocPrint(
+        allocator,
+        "Hello, {s}!",
+        .{name},
+    );
+}
+
+// _greet prints a greeting to the console.
+pub fn _greet(name: []const u8) void {
+    const g = _greeting(name);
+    const s = std.fmt.allocPrint(
+        allocator,
+        "wasm >> {s}",
+        .{g},
+    ) catch unreachable;
```

Comment:
why are we catching unreachable? if there's an allocation failure we should fail all the way out, right? none of the other examples swallow errors.

one thing to keep in mind is if we are doing something in the example, it is indeed an example, so we are saying this is the right way to code. If catching unreachable is normal, then here is the right thing, though probably needs a line comment or in the README why we catch unreachable.

Reply:
`catch unreachable` here in zig means `if there's an error, panics`, so this doesn't swallow!

Reply:
> `catch unreachable` here in zig means `if there's an error, panics`, so this doesn't swallow!

This is incorrect. That's only the [case for safe building](https://ziglang.org/documentation/master/#unreachable) modes such as debug and release-safe. Considering we're building for Wasm, people tend to use release-small to get the smallest binary size. In such a case, this would lead into undefined behavior. `unreachable` should only be used when the author knows it *should* never be hit, unless a developer makes a mistake.

The correct way to solve this is to handle the error appropriately and notify the runtime. For example purposes, you could also use `@panic()` instead.

Reply:
@Luukdegram Thanks for your input. I didn't know `unreachable` behaved differently depending on modes.
`@panic()` takes `[]u8` and I had no idea what is an idiomatic implementation here, so I just switched and returned `[]u8` according to the returned error. Please let me know if there is a better way.
https://github.com/tetratelabs/wazero/pull/741/commits/398dc4f0bf6bc056dd044e52fc4e9e9b8f082573